src/backend/app.js

```diff
@@ -20,6 +20,89 @@ app.use(express.json());
 app.use(cors());
 
 router.use("/meals", mealsRouter);
+router.get('/', (req, res) => {
+  res.send('Hi friend');
+});
+//Respond with all meals in the future
+//(relative to the when datetime):
+router.get('/future-meals', async (req, res) => {
+  try {
+    //const NOW = new Date();
+    const futureMeals = await knex.raw(
+      `
+  SELECT * FROM meals
+   WHERE date > NOW()`
+    );
+    res.json(futureMeals);
+  } catch (error) {
+    throw error;
+  }
+});
+//Respond with all meals in the past
+//(relative to the when datetime):
+router.get('/past-meals', async (req, res) => {
+  try {
+    //const now = new Date();
+    const pastMeals = await knex.raw(
+      `
+  SELECT * FROM meals
+   WHERE date < NOW()`
+    );
+
+    router.use('/meals', mealsRouter);
+    res.json(pastMeals);
+  } catch (error) {
+    throw error;
+  }
+});
+//Respond with all meals sorted by ID:
+router.get('/all-meals', async (req, res) => {
+  try {
+    const allMeals = await knex.raw(
+      `
+  SELECT * FROM meals
+  ORDER BY idmeals
+   `
+    );
+    res.json(allMeals);
+  } catch (error) {
+    throw error;
+  }
+});
+//Respond with the first meal (meaning with the minimum id):
+router.get('/first-meals', async (req, res) => {
+  try {
+    const firstMeals = await knex.raw(
+      `
+  SELECT * FROM meals
+  WHERE idmeals=( SELECT MIN(idmeals) FROM  meals)
+   `
+    );
+    if (firstMeals.length === 0) {
+      res.status(404).json({ error: 'There are no meals' });
```

Comment:
Choose status 400 instead of 404. Here is the difference:
- 400 Bad Request: The server was unable to process the request due to a client error, such as invalid syntax or missing parameters.
- 404 Not Found: The requested resource was not found on the server. This may be due to a misspelled URL, a deleted or moved resource, or incorrect permissions.

Reply:
thanks so much for the review i will  modify my code